.github/workflows/verify_docker_image.yml

```
@@ -27,7 +93,22 @@ jobs:
           --group-add video \
           --cap-add=SYS_PTRACE \
           -w /TheRock \
-          ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_gfx942:main \
+          ${{ env.IMAGE_REF }} \
           bash external-builds/pytorch/smoke-tests/run_smoke_tests.sh
+
       - name: Verify Containers Status
-        run: docker ps -a
+        run: |
+          docker ps -a
+      # Tag and push verified image
+      - name: Tag and push verified image
+        run: |
+          IMAGE=ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_${{ inputs.amdgpu_target }}
+          DIGEST=${{ inputs.image_digest  }}
+          VERSION=${{ inputs.version }}
+
+          echo "Using image: $IMAGE"
+          echo "Using digest: $DIGEST"
+          echo "Using version: $VERSION"
+
+          docker tag $IMAGE@$DIGEST $IMAGE:$VERSION
```

I think we probably want to reuse the environment variable here as well?
```suggestion
          docker tag ${{ env.IMAGE_REF }} $IMAGE:$VERSION
```